.github/workflows/main.yml

```diff
@@ -0,0 +1,273 @@
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-
+
+      - name: Cache node version manager
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-nvm-dh
+        with:
+          path: ~/.nvm/versions/
+          key: ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-npm-modules-dh
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: sudo apt -y install libcurl4-openssl-dev libssl-dev pkg-config
+
+      - name: Install pip requirements
+        run: |
+          mkdir ./.cache && mkdir ./.cache/pip
+          chown -R $(whoami) ./.cache/pip
+          pip install -U pip setuptools wheel --cache-dir ./.cache/pip
+          pip install -r requirements/main.txt --cache-dir ./.cache/pip
+          pip install -r requirements/tests.txt --cache-dir ./.cache/pip
+          pip install -r requirements/deploy.txt --cache-dir ./.cache/pip
+          
+      - name: Setup node
+        run: |
+          curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
+          export NVM_DIR="$HOME/.nvm" 
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+          [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
+          nvm install $NODE_VERSION
+          npm i -g npm
+          npm install -g gulp-cli
+          npm ci
+
+      - name: Run tests
+        run: |
+          python -m coverage run -m pytest --strict --postgresql-host localhost
+          python -m coverage html
+          python -m coverage report -m --fail-under 100
+        env:
+          # The hostname used to communicate with the PostgreSQL service container
+          POSTGRES_HOST: localhost
+          # The default PostgreSQL port
+          POSTGRES_PORT: 5432
```

Comment:
Are these used? The test command above is specifying `localhost` not `POSTGRES_HOST`. Also, are these necessary if they're the defaults?

Reply:
Yep, not necessary, was using this in a previous iteration when I was trying different hosts/ports. Good catch.